### ISSUE TITLE
Modules using git submodules

### ISSUE THREAD

**Issue by gibfahn** [nodejs/citgm#300]
As it's come up a couple of times, we might as well have a separate issue to discuss testing npm modules that use git submodules, not least so we have something to link to when the discussion comes up again. 

The current stance (as I understand it) is that modules should be entirely available as a Github tarball in order to be tested through CitGM in the Node community. This means that if an npm module normally uses submodules, in order to qualify its npm releases must correspond to a Github tag containing a checked-in version of all the submodules.

Related issues:
https://github.com/nodejs/citgm/issues/241
https://github.com/nodejs/citgm/pull/292
https://github.com/nodejs/citgm/pull/240
https://github.com/nodejs/citgm/pull/159
https://github.com/nodejs/citgm/issues/187

**Comment by MylesBorins** [comment]
TLDR we are defaulting to using the tarball served from github. Submodules cannot work unless you are working from a git repo.

We could potentially shallow clone specific tags rather than tarballs and then install submodules.... this adds quite a bit of complexity though and I have serious mixed feelings about it

**Comment by bzoz** [comment]
Passing github repo to `npm pack` will also fetches submodules. Calling `node citgm.js https://github.com/redis/hiredis-node` works as expected, we could use this.

**Comment by MylesBorins** [comment]
> Passing github repo to npm pack will also fetches submodules.

This isn't exactly what is going on. The publishers of `hiredis-node` are squashing their submodules prior to cutting tags, which is why the submodule is available in the tarball. This is actually the suggested way to deal with tarballs if projects cannot avoid using them.

**Comment by richardlau** [comment]
git submodules are now supported by https://github.com/nodejs/citgm/pull/695.
